sympy/functions/combinatorial/factorials.py

```diff
@@ -849,10 +890,83 @@ def eval(cls, n, k):
             from sympy import gamma
             return gamma(n + 1)/(gamma(k + 1)*gamma(n - k + 1))
 
+    def _eval_Mod(self, q):
+        n, k = self.args
+        if all(x.is_integer for x in (n, k, q)):
+            if all(x.is_Integer for x in (n, k, q)):
+                n, k = map(int, (n, k))
+                aq = abs(q)
+
+                # handle negative integers k or n
+                if k < 0:
+                    return 0
+                if n < 0:
+                    n = -n + k - 1
```

Comment:
Should the sign be recorded?

Reply:
Yes, this is a bug, ans would be positive/negative depending on parity of k, so sign should be recorded, will make the changes.